Lern-API/Models/User.cs

```diff
@@ -1,32 +1,36 @@
-﻿using System.Diagnostics.CodeAnalysis;
-using FluentValidation;
+﻿using System;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text.Json.Serialization;
 
 namespace Lern_API.Models
 {
-    public class User : AbstractModel
+    public class User : IModelBase
     {
-        public string Name { get; set; }
+        [ReadOnly(true), Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public Guid Id { get; set; }
+        [ReadOnly(true)]
+        public DateTime CreatedAt { get; set; }
+        [ReadOnly(true)]
+        public DateTime UpdatedAt { get; set; }
+        [ReadOnly(true)]
+        public User Manager { get; set; }
+        public string Firstname { get; set; }
+        public string Lastname { get; set; }
+        public string Nickname { get; set; }
```

Comment:
Incoherence in naming: User creation route uses "nickname" while login route uses "login"  for the same field

Reply:
`/login` route accepts both nickname and email as a valid input to authenticate a user, hence the naming